tools/SourceKit/lib/SwiftLang/SwiftSourceDocInfo.cpp

```diff
@@ -755,7 +755,7 @@ static bool passCursorInfoForDecl(const ValueDecl *VD,
         PO.SkipIntroducerKeywords = true;
         PO.ArgAndParamPrinting = PrintOptions::ArgAndParamPrintingMode::ArgumentOnly;
         XMLEscapingPrinter Printer(RelDeclsStream);
-        if (BaseType) {
+        if (BaseType && RelatedDecl->getDeclContext() == BaseType->getAnyNominal()) {
```

Comment:
Is the base type ever different from the type of the declaration's context? E.g. If the decl is a method on `G<T>` can the base type be `G<ain't>`? If not you could simplify things and get rid of the base type altogether.